src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PEFieldSymbol.cs

```diff
@@ -637,7 +637,7 @@ internal override UseSiteInfo<AssemblySymbol> GetUseSiteInfo()
                 UseSiteInfo<AssemblySymbol> result = new UseSiteInfo<AssemblySymbol>(primaryDependency);
                 CalculateUseSiteDiagnostic(ref result);
                 if (RefKind != RefKind.None &&
-                    (IsFixedSizeBuffer || Type.IsRefLikeType == true))
+                    (IsFixedSizeBuffer || Type.IsRefLikeType == true || !ContainingType.IsRefLikeType))
```

Comment:
Is there a safety hole if we permit _any_ usage of such a type? For example, if we allowed using the containing type as a type parameter?

Reply:
Do we have the same issue today with `private` fields (that are not imported when the containing type is a reference type)?

Reply:
That's a fair point, but until now there hasn't been a runtime that supported ref fields so it's been a bit of a moot question. Now that there is, is there a problem?